### ISSUE TITLE
Typescript Compile Errors 'args' implicitly has 'any[]' type

### ISSUE THREAD

**Issue by bpred754** [jeremydaly/serverless-mysql#108]
When I run `tsc` on project I receive the following errors:

```
node_modules/serverless-mysql/index.d.ts:93:11 - error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.

93     query(...args): this
             ~~~~~~~

node_modules/serverless-mysql/index.d.ts:101:14 - error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.

101     query<T>(...args): Promise<T>
```

This is requiring us to use `skipLibCheck` compiler option which is not ideal.

**Comment by levino** [comment]
This comes from violating [best practices](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html).
>If your types are generated by your source code, publish the types with your source code. Both TypeScript and JavaScript projects can generate types via --declaration.
>
>**Otherwise**, we recommend submitting the types to DefinitelyTyped, which will publish them to the @types organization on npm.

Since the author of this package is not a TypeScript coder, they should also not be burdened with the task of writing and maintaining types manually (which is quite hard). Also there are no tests here for the types, so bugs will be tedious to find and regressions impossible to prevent.

I suggest to move these types to DT where they belong. I created a [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54073) already. I would say we need some more tests over there, will continue on Monday probably. The types here should be removed from the repo asap (DT tslint check does not like them very much :D). Until then I will monkey patch this file out with [patch-package](https://www.npmjs.com/package/patch-package).

Also  this is related to #109.

**Comment by naorpeled** [comment]
Hey,
the fix for this is included in v1.5.5 🙏
